controllers/appsv1/pod_webhook.go

```diff
@@ -0,0 +1,217 @@
+package appsv1
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/viper"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	v1 "github.com/jaegertracing/jaeger-operator/apis/v1"
+	"github.com/jaegertracing/jaeger-operator/pkg/inject"
+)
+
+var (
+	_ admission.DecoderInjector = (*podInjector)(nil)
+	_ webhook.AdmissionHandler  = (*podInjector)(nil)
+)
+
+// NewPodInjectorWebhook creates a new pod injector webhook to be registered
+func NewPodInjectorWebhook(c client.Client) webhook.AdmissionHandler {
+	return &podInjector{
+		client: c,
+	}
+}
+
+// You need to ensure the path here match the path in the marker.
+// +kubebuilder:webhook:path=/mutate-v1-pod,mutating=true,failurePolicy=fail,groups="",resources=pods,sideEffects=None,verbs=create,versions=v1,name=object.sidecar-injector.jaegertracing.io,admissionReviewVersions=v1;v1beta1
```

Comment:
why this line is repeated 3 times? As I can see only the `name` parameter changes. Why? 

Could we use only `v1` for `admissionReviewVersions` ? In what k8s versions the v1 was added?

Reply:
> why is this line repeated 3 times? As I can see only the name parameter changes. Why?

There are 3 different patches. Seems since we use the webhook for `create`, `update` and `delete` sidecars, its not needed. 

```suggestion
// +kubebuilder:webhook:path=/mutate-v1-pod,mutating=true,failurePolicy=fail,groups="",resources=deployments;pods,sideEffects=None,verbs=create;update;delete,versions=v1,name=object.sidecar-injector.jaegertracing.io,admissionReviewVersions=v1
```

> Could we use only v1 for admissionReviewVersions ? In what k8s versions the v1 was added?

yes i changed it :+1: `API version v1, a version introduced in Kubernetes v1.16`.

Reply:
Wait, it doesnt make sense to add deployments there, right? Because those deployments create a pod anyway.